src/stream.ts

```diff
@@ -97,7 +98,27 @@ export class WebRTCStream implements Stream {
         res = new Uint8Array(data as ArrayBuffer);
       }
       log.trace(`[stream:${this.id}][${this.stat.direction}] received message: length: ${res.length} ${res}`);
-      (this.source as Pushable<Uint8ArrayList>).push(new Uint8ArrayList(res));
+      let m = pb.Message.fromBinary(res);
+      log(`[stream:${this.id}][${this.stat.direction}] received pb.Message: ${Object.entries(m)}`);
+      switch (m.flag) {
+        case undefined:
+          break; //regular message only
+        case pb.Message_Flag.STOP_SENDING:
+          log.trace('Remote has indicated, with "STOP_SENDING" flag, that it will discard any messages we send.');
+          this.closeWrite();
+          break;
+        case pb.Message_Flag.FIN:
+          log.trace('Remote has indicated, with "FIN" flag, that it will not send any further messages.');
+          this.closeRead();
+          break;
+        case pb.Message_Flag.RESET:
+          log.trace('Remote abruptly stopped sending, indicated with "RESET" flag.');
+          this.closeRead();
```

Comment:
Is there the possibility that this isn't exhaustive and needs a default case?

Reply:
What would a default case do? Throw an exception?

Reply:
I don't know enough of the context to know, but I guess the `undefined` case is acting like the default case.  I was just thinking of the possibility of other values not here, like `null` or whatever else is possible.  I think you can ignore this.

Reply:
`undefined` is there because the flag is optional in the protobuf. If it's missing it means it's a normal packet doing normal streaming things for the application (the protocol allows for the flag and message to both be present, but in practice it's one or the other).

In Javascript if it's a value that's not handled and there's no default, does it skip the whole switch like it would in C ? If so I think that's very reasonable behavior if someone tries to throw null at us somehow (presumably a coding error).

Reply:
Following up here, I think the implementation should support all possible cases:

- no data and no flag: Just ignore the hole packet. Could e.g. be possible in case we extend the Protobuf schema in a backwards compatible way in the future.
- data and no flag
- data and flag
- no data and flag

Reply:
@mxinden The `switch` and `if` seem to be independent to me, so the behavior you're describing is what I believe to already be the case.

For example, the `FIN` `case` in `switch` does a `break` not a `return`, so after handling that you head on to the `if` check for `message`

But if you got undefined/undefined you'd break out of the switch without doing anything and then pass over the if.